### PR TITLE
perf(hy_v4): add optional AITER paged MQA logits

### PR DESCRIPTION
**Pull request by alexanderbin123**
## Problem

HY V4 decode currently computes FP8 paged-MQA logits through the existing fallback before TopK. AITER now exposes a page-size-64 `paged_mqa_logits` entry point that can consume the native HY V4 FP8 indexer cache directly.

## Change

- Add opt-in `VLLM_HCU_USE_AITER_OPUS_PAGED_MQA_LOGITS=1` routing on gfx938.
- Reuse the native `[pages, 64, 1, 132]` FP8+scale cache and block table without repacking.
- Restrict routing to supported FP8 dtype, head shapes, MTP widths, contiguous metadata, and sufficient block-table capacity.
- Keep padded/unsupported decode batches on the existing implementation.
- Let the AITER route take precedence over the LightOp mask producer when both flags are enabled.

This PR is stacked on #131. The new route is disabled by default.

## Validation

- Focused suite: `199 passed`.
- Numerical reference: max abs error `2.288818e-05`, mean abs error `1.280357e-06`, mean relative error `3.22128e-08`, TopK overlap `100%`.
- Controlled operator benchmark, 16 requests/card, MTP3 (`R=4`), context 4096, TopK 2048, CUDA Graph:
  - AITER paged-MQA + TopK: `48.80 us`
  - LightOp sparse-mask pair: `110.40 us`
  - Decode indexer pair latency: `-55.8%` (`2.26x`).
- DP8+EP8, DeepEP low latency, MTP3, CUDA Graph, FP8 KV dequant:
  - service startup and graph capture passed;
  - smoke request HTTP 200;
  - HumanEval limit 8: Accuracy `100%`, Pass@1 `100%`;
  - average latency `6.956 s/request`, output throughput `17.67 tok/s`;
  - MTP acceptance `745/783 = 95.15%`.

The 8-sample HumanEval latency is effectively flat against the prior `6.921 s/request`; the measurable gain is isolated to the paged-MQA + TopK operator pair.

### CODE REVIEW

**Comment by alexanderbin123** [comment]
Validated server command (DeepEP `1.1.0+dtk2604.torch2110.2609171757.gc807df`):

```bash
export PYTHONPATH=/path/to/vllm-plugin-das${PYTHONPATH:+:$PYTHONPATH}
export HIP_VISIBLE_DEVICES=0,1,2,3,4,5,6,7
export VLLM_USE_V2_MODEL_RUNNER=1
export VLLM_HCU_USE_CUSTOM_FLASH_ATTN=1
export VLLM_HCU_HYV4_FP8_KV_DEQUANT=1
export VLLM_HCU_USE_LIGHTOP_SPARSE_MLA_TOPK=1
export VLLM_HCU_USE_LIGHTOP_MASK_TOPK=0
export VLLM_HCU_USE_LIGHTOP_FAST_TOPK_TRANSFORM=1
export VLLM_HCU_USE_AITER_OPUS_PAGED_MQA_LOGITS=1
export TORCH_BLAS_PREFER_CUBLASLT=1
export ROCSHMEM_HEAP_SIZE=4000000000
export VLLM_DP_MASTER_IP=127.0.0.1
export VLLM_DP_MASTER_PORT=29671
unset VLLM_PLUGINS HTTP_PROXY HTTPS_PROXY ALL_PROXY http_proxy https_proxy all_proxy
export NO_PROXY=127.0.0.1,localhost
export no_proxy=127.0.0.1,localhost

python -m vllm.entrypoints.cli.main serve /models/Hy4-preview-Channel-FP8-w8a8 \
  --host 127.0.0.1 \
  --port 20134 \
  --served-model-name hy4-dp8-ep8-mtp3-aiter-opus \
  --trust-remote-code \
  --dtype bfloat16 \
  -q compressed-tensors \
  --tensor-parallel-size 1 \
  --data-parallel-size 8 \
  --enable-expert-parallel \
  --all2all-backend deepep_low_latency \
  --moe-backend deep_gemm \
  --kv-cache-dtype fp8_ds_mla \
  --block-size 64 \
  --max-model-len 8192 \
  --max-num-seqs 16 \
  --max-num-batched-tokens 256 \
  --gpu-memory-utilization 0.95 \
  --kv-cache-memory-bytes 536870912 \
  --enable-prefix-caching \
  --default-chat-template-kwargs '{"reasoning_effort":"no_think"}' \
  --reasoning-parser hy_v4 \
  --enable-auto-tool-choice \
  --tool-call-parser hy_v4 \
  --speculative-config '{"method":"mtp","num_speculative_tokens":3}' \
  --cudagraph-capture-sizes 1 2 4 8 16 32 64 \
  --data-parallel-rpc-port 29672
```

`VLLM_HCU_USE_LIGHTOP_MASK_TOPK=0` is intentional for this configuration: the AITER paged-MQA producer is selected, while LightOp still handles decode TopK. Validation: CUDA Graph capture passed, smoke HTTP 200, HumanEval 8/8, MTP3 acceptance 95.15%.
